src/DynamoCoreWpf/ViewModels/Core/Converters/SerializationConverters.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Linq;
+using Dynamo.Graph.Annotations;
+using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Notes;
+using Dynamo.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Type = System.Type;
+
+namespace Dynamo.Wpf.ViewModels.Core.Converters
+{
+    /// <summary>
+    /// The AnnotationConverter is used to serialize and deserialize AnnotationModels.
+    /// The SelectedModels property on AnnotationModel is a list of references
+    /// to ModelBase objects. During serialization we want to refer to these objects
+    /// by their ids. During deserialization, we use the ReferenceResolver to
+    /// find the correct ModelBase instances to reference.
+    /// </summary>
+    public class AnnotationConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(AnnotationModel);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var obj = JObject.Load(reader);
+            var title = obj["Title"].Value<string>();
+            //if the id is not a guid, makes a guid based on the id of the model
+            Guid annotationId = GuidUtility.tryParseOrCreateGuid(obj["Id"].Value<string>());
+
+            // This is a collection of string Guids, which
+            // should be accessible in the ReferenceResolver.
+            var models = obj["Nodes"].Values<JValue>();
+
+            var existing = models.Select(m =>
+            {
+                Guid modelId = GuidUtility.tryParseOrCreateGuid(m.Value<string>());
+
+                return serializer.ReferenceResolver.ResolveReference(serializer.Context, modelId.ToString());
+            });
+
+            var nodes = existing.Where(m => typeof(NodeModel).IsAssignableFrom(m.GetType())).Cast<NodeModel>();
+            var notes = existing.Where(m => typeof(NoteModel).IsAssignableFrom(m.GetType())).Cast<NoteModel>();
+
+            var anno = new AnnotationModel(nodes, notes);
+            anno.AnnotationText = title;
+            anno.GUID = annotationId;
+            anno.X = obj["Left"].Value<double>();
+            anno.Y = obj["Top"].Value<double>();
+            anno.Width = obj["Width"].Value<double>();
+            anno.Height = obj["Height"].Value<double>();
+            anno.FontSize = obj["FontSize"].Value<double>();
+            anno.InitialTop = obj["InitialTop"].Value<double>();
+            anno.InitialHeight = obj["InitialHeight"].Value<double>();
+            anno.TextBlockHeight = obj["TextblockHeight"].Value<double>();
+            anno.Background = obj["Background"].Value<string>();
+            return anno;
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var anno = (AnnotationModel)value;
```

Comment:
I feel it would be better if this Writer serialize properties on `AnnotationViewModel`? To achieve that I suggest:
1. Override `CanConvert` to look for `AnnotationViewModel` 
2. Remove the `JsonIgnore`s on `AnnotationViewModel`
3. var anno = (AnnotationViewModel)value;